client/my-sites/media/main.jsx

```diff
@@ -268,10 +272,28 @@ class Media extends Component {
 		MediaActions.delete( site.ID, selected );
 	};
 
+	getAnalyticsPath = () => {
+		const { filter, selectedSite } = this.props;
```

Comment:
Maybe let's invert this a bit, as a bonus it's a little easier to search for in an IDE:
```jsx
if ( selectedSiteId && filter ) { 
    //hmm what happens if the site isn't loaded yet? We pass an incorrect path to the page tracker component right?
    // a getSiteFragment check may be more appropriate
    return `/media/${ filter }/:site`
}

if ( selectedSiteId ) {
    return `/media/:site`;
}

return '/media';
```

Reply:
Afaik @Copons introduced a `hasSelectedSiteLoaded` check in `PageViewTracker` that should help guard against this case.